corehq/ex-submodules/pillowtop/reindexer/reindexer.py

```diff
@@ -188,7 +185,8 @@ def clean(self):
     def reindex(self):
         if not self.in_place and not self.start_from:
             _prepare_index_for_reindex(self.es, self.index_info)
-            _set_checkpoint(self.pillow)
+            if isinstance(self.pillow_or_processor, ConstructedPillow):
```

Comment:
how will checkpointing happen in the reindexers now?

Reply:
I think there are two instances of reindexing.
1. **When updating any index** We have discussed this a bit and decided that when we change an index (say case index), we add a new pillow for the new index and let the pillow with new index catch up with the combined pillow before replacing the index and deploy. We have decided to keep the old pillows around for only this purpose.
Though this means, I should revert from passing just the processor to passing the whole of old pillow with single processors, which I missed and can do now. More context [here](https://docs.google.com/a/dimagi.com/document/d/19GjOvfVfaSj5q10pueM5fb9cVy0-VsbbNJ-HZGnL7fo/edit?disco=AAAACTlKlow)
2. **Doing a full ptop_reindexer_v2** Assuming this is only done in development, this won't be an issue. If we do want to make this work in production, we can create new reindexers for combined pillows.